### ISSUE TITLE
Included resources in SchematronResourceSCH are not released after validation

### ISSUE THREAD

**Issue by monapadu** [phax/ph-schematron#110]
Hi,
When using SchematronResourceSCH with a schematron with includes, after running the validation, the included resources are not being released and cannot be deleted.
```
java.nio.file.FileSystemException: c:\sch\abstract\XRechnung-model.sch: The process cannot access the file because it is being used by another process.

        at sun.nio.fs.WindowsException.translateToIOException(Unknown Source) ~[?:1.8.0_172]
        at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:1.8.0_172]
        at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source) ~[?:1.8.0_172]
        at sun.nio.fs.WindowsFileSystemProvider.implDelete(Unknown Source) ~[?:1.8.0_172]
        at sun.nio.fs.AbstractFileSystemProvider.delete(Unknown Source) ~[?:1.8.0_172]
        at java.nio.file.Files.delete(Unknown Source) ~[?:1.8.0_172]
```
```   
  @Test(expected = FileSystemException.class)
  public void testIncludedFilesNotDeleted() throws Exception
  {
    SchematronResourceSCH.fromClassPath ("test-sch/atgov/ATGOV-UBL-T10.sch");
    final ISchematronResource aSV = SchematronResourceSCH.fromClassPath ("test-sch/atgov/ATGOV-UBL-T10.sch");
    aSV.applySchematronValidation (new ClassPathResource (VALID_XMLINSTANCE));
    Files.delete(Paths.get(ClassLoader.getSystemResource("test-sch/atgov/include/ATGOV-T10-abstract.sch").toURI()));
    Files.delete(Paths.get(ClassLoader.getSystemResource("test-sch/atgov/include/ATGOV-T10-abstract.sch").toURI()));
  }
```

**Comment by phax** [comment]
Thanks @monapadu  for pointing that out 👍

**Comment by phax** [comment]
That seems to be a problem in Saxon. In class "DocumentFn" this code opens an input stream without closing it:

```java
        if (source instanceof StreamSource && ((StreamSource)source).getInputStream() == null && ((StreamSource)source).getReader() == null) {
            String uri = source.getSystemId();
            if (uri != null) {
                resolver = context.getController().getStandardURIResolver();
                try {
                    source = resolver.resolve(uri, "");
                } catch (TransformerException ex) {
                    throw XPathException.makeXPathException(ex);
                }
            } else {
                source = null; // the supplied source is useless, treat it as if the URIResolver had returned null
            }
        }
```

**Comment by phax** [comment]
A respective issue has been opened: https://saxonica.plan.io/issues/4833

**Comment by monapadu** [comment]
Thanks for investigating and opening the issue in saxon!

**Comment by phax** [comment]
This requires ph-commons 10.0 so we need to wait a bit...

**Comment by phax** [comment]
This was fixed in ph-schematron 6.1.0
